server/cp/macros/translate_to_desk_language.py

```diff
@@ -28,8 +34,21 @@ def translate_to_desk_language(item, **kwargs):
             item['language'] = 'fr-CA'
         elif dest_desk.get('desk_language') == 'en-CA' and item.get('language') != 'fr-CA':
             item['language'] = 'en-CA'
-    elif kwargs['desk'].get('desk_language') == 'fr-CA':
+    elif dest_desk and dest_desk.get('desk_language') == 'fr-CA':
         item['language'] = 'fr-CA'
+    elif dest_desk_id:
+        desk = get_resource_service('desks').find_one(req=None, _id=dest_desk_id)
+        if desk and desk.get('desk_language') == 'fr-CA':
+            item['language'] = 'fr-CA'
+    else:
+        try:
+            current_desk_id = item.get('task', {}).get('desk')
+            desk = get_resource_service('desks').find_one(req=None, _id=current_desk_id)
+            if desk.get('desk_language') == 'fr-CA':
+                item['language'] = 'fr-CA'
+        except KeyError as err:
```

Comment:
not sure where you could get that `KeyError` here if you use `.get` always, maybe it should be another exception?